apps/public-docsite-v9/src/Concepts/Slots/Slots.stories.mdx

```diff
@@ -0,0 +1,365 @@
+            ? WithSlotRenderFunction<Props>
+            : Type
+        >
+      ...
+```
+
+Following `WithSlotShorthandValue<Props>`, it is defined as the `Props` object
+or the `children` of a React child, array of nodes, or portal.
+
+```ts
+type WithSlotShorthandValue<Props extends { children?: unknown }> =
+  | Props
+  | Extract<SlotShorthandValue, Props['children']>;
+
+type SlotShorthandValue = React.ReactChild | React.ReactNodeArray | React.ReactPortal;
+```
+
+Following `WithSlotRenderFunction<Props>`, it is defined as the `Props` object and children.
+The children can be JSX/TSX or a function that renders JSX/TSX.
+
+```ts
+type WithSlotRenderFunction<Props extends { children?: unknown }> = Props & {
+  children?: Props['children'] | SlotRenderFunction<Props>;
+};
+
+type SlotRenderFunction<Props> = (
+  Component: React.ElementType<Props>,
+  props: Omit<Props, 'children' | 'as'>,
+) => React.ReactNode;
+```
+
+## Component slot usage
+
+### Defining slots
+
+In the Fluent UI React composition architecture, components define their props using `ComponentProps<Slot>`
+and state for rendering using `ComponentState<Slot>`.
+
+For example, Button defines an icon slot.
+
+```tsx
+type ButtonSlots = {
+  root: NonNullable<Slot<ARIAButtonSlotProps>>;
+  icon?: Slot<'span'>;
+};
+
+type ButtonProps = ComponentProps<ButtonSlots> & {
+  ...
+};
+
+type ButtonState = ComponentState<ButtonSlots> & {
+  ...
+};
+}
+```
+
+### The special root slot
+
+Every component has a `root` slot.
+You can think of the root slot as
+the element the component renders that contains all component's rendered content.
+
+The root slot
+
+- defines the type of element that will be rendered
+- always gets `className` and `style` properties
+- get the properties passed to the component
+
+### NonNullable slots
+
+When a slot is defined in the props of a component, it can be set to null to prevent the slot from being rendered.
+In cases where components need to require a slot to always be rendered, it can be wrapped in `NonNullable<T>`.
+
+For example, in `RadioButton` the indicator must always be rendered.
+
+```ts
+indicator: NonNullable<Slot<'div'>>;
+```
+
+### Styling and Rendering with slots
+
+You can think of each slot as container for props and style that will be rendered as an element.
+
+The Fluent UI hooks architecture breaks up component rendering primarily into 3 parts
+
+- use*Component*(): This takes in props and produces state
+- use*Component*Styles() - This uses state to define and apply class styles
+- render*Component*() - This renders the elements of the component
+
+`Button` follows this pattern:
+
+- `useButton` will take in `ButtonProps` and return `ButtonState`.
+
+```ts
+const useButton_unstable = (
+  props: ButtonProps,
+  ref: React.Ref<HTMLButtonElement | HTMLAnchorElement>,
+): ButtonState
+```
+
+Hooks like `useButton` will typically fill in default values, calculate state from props, add event handlers,
```

Comment:
```suggestion
Hooks like `useButton` calculate state from props. Typically, this involves filling in default values, adding event handlers, 
```

This sentence is broken up to two lines so the next line will need to be changed if you accept this edit.